node/impl/storminer.go

```diff
@@ -97,6 +98,9 @@ type StorageMinerAPI struct {
 	Epp gen.WinningPoStProver `optional:"true"`
 	DS  dtypes.MetadataDS
 
+	// StorageService is populated when we're not the main storage node (e.g. we're a markets node)
+	StorageService modules.MinerStorageService `optional:"true"`
```

Comment:
How does this get set.  I'm guessing there's DI magic but I don't see the field being set or an explicit config to initialize this which I kind of expect since its optional.

Reply:
The StorageMinerAPI struct is filled through fx.Extract, with each field being field individually thanks to the `fx.In` "trait" on the struct. `optional` tells Fx to not fail if a constructor for `modules.MinerStorageService` doesn't exist in the DI container - instead it leaves it not set (nil).